pkg/ingester/ingester.go

```diff
@@ -480,8 +480,22 @@ func (i *Ingester) checkRunningOrStopping() error {
 	return status.Error(codes.Unavailable, s.String())
 }
 
+// Using block store, the ingester is only available when it is in a Running state. The ingester is not available
+// when stopping to prevent any read or writes to the TSDB after the ingester has closed them.
+func (i *Ingester) checkRunning() error {
+	s := i.State()
+	if s == services.Running {
+		return nil
+	}
+	return status.Error(codes.Unavailable, s.String())
+}
+
 // Push implements client.IngesterServer
 func (i *Ingester) Push(ctx context.Context, req *cortexpb.WriteRequest) (*cortexpb.WriteResponse, error) {
+	if i.cfg.BlocksStorageEnabled {
+		return i.v2Push(ctx, req)
+	}
+
```

Comment:
For `Push` it's ok to `CheckRunning()` in both cases, that will avoid duplicating the other checks.
There is a comment to say something else will catch it, but it doesn't matter.  We don't want to accept writes while stopping.

Reply:
Makes sense. I've updated it so it calls `checkRunning()` in both cases.